kaldb/src/main/java/com/slack/kaldb/server/Kaldb.java

```diff
@@ -394,8 +394,10 @@ private static Set<Service> getServices(
               KaldbConfigs.NodeRole.PREPROCESSOR, List.of(datasetMetadataStore)));
 
       if (preprocessorConfig.getUseBulkApi()) {
-        armeriaServiceBuilder.withAnnotatedService(
-            new OpenSearchBulkIngestApi(datasetMetadataStore, preprocessorConfig, meterRegistry));
+        OpenSearchBulkIngestApi openSearchBulkApiService =
+            new OpenSearchBulkIngestApi(datasetMetadataStore, preprocessorConfig, meterRegistry);
+        armeriaServiceBuilder.withAnnotatedService(openSearchBulkApiService);
+        services.add(openSearchBulkApiService);
```

Comment:
Let's add a todo here to revisit this - I'm not sure I'm a huge fan of mixing the Armeria service with a guava service in the same class.